go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -158,25 +157,32 @@ func routeToEngineRoute(ctx *plancontext.PlanningContext, op *operators.Route) (
 	if err != nil {
 		return nil, err
 	}
-	var vindex vindexes.Vindex
-	var values []evalengine.Expr
-	if op.SelectedVindex() != nil {
-		vindex = op.Selected.FoundVindex
-		values = op.Selected.Values
+
+	rp := newRoutingParams(ctx, op.Routing.OpCode())
+	err = op.Routing.UpdateRoutingParams(ctx, rp)
+	if err != nil {
+		return nil, err
 	}
+
 	return &engine.Route{
-		TableName: strings.Join(tableNames, ", "),
-		RoutingParameters: &engine.RoutingParameters{
-			Opcode:              op.RouteOpCode,
-			Keyspace:            op.Keyspace,
-			Vindex:              vindex,
-			Values:              values,
-			SysTableTableName:   op.SysTableTableName,
-			SysTableTableSchema: op.SysTableTableSchema,
-		},
+		TableName:         strings.Join(tableNames, ", "),
+		RoutingParameters: rp,
 	}, nil
 }
 
+func newRoutingParams(ctx *plancontext.PlanningContext, opCode engine.Opcode) *engine.RoutingParameters {
+	ks, _ := ctx.VSchema.DefaultKeyspace()
+	if ks == nil {
+		// if we don't have a selected keyspace, any keyspace will do
+		// this is used by operators that do not set the keyspace
+		ks, _ = ctx.VSchema.AnyKeyspace()
+	}
```

Comment:
should we not check here for the opcodes which are allowed to use `AnyKeyspace`?

Reply:
This is just setting the default keyspace. Routings will then set the keyspace they will use, if they specify if.